### PR TITLE
now we render the data we request from github

### PR DESCRIPTION
**Pull request by izelnakri**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
     method: 'GET',
-    url: 'https://api.github.com/users/izelnakri',
+    url: 'https://api.github.com/users/' + req.params.username,
     headers: {
       'User-Agent': 'Izels awesome JS Code'
     }
   }, (error, response, body) => {
     if (!error) {
-      res.json(JSON.parse(body));
+      res.render('users/show', { user: JSON.parse(body) });
```
Comment (izelnakri) [inline index.js:25]: what is body value here in JavaScript?

File: index.js [hunk 2]
```diff
@@ -1,25 +1,28 @@
...
 app.get('/', (request, response) => {
   response.redirect('/profile');
 });
 
-app.get('/profile', (req, res) => {
+app.get('/:username', (req, res) => {
   request({
     method: 'GET',
-    url: 'https://api.github.com/users/izelnakri',
+    url: 'https://api.github.com/users/' + req.params.username,
```
Comment (izelnakri) [inline index.js:19]: what is req.params.username?

Comment (Saidac) [reply index.js:19]: require parameter username from key?

Comment (ghost) [reply index.js:19]: requesting the headers/content of the given username from the github server?

Comment (Zahra120) [reply index.js:19]: It is the dynamic part that we put in the url.

File: index.js [hunk 3]
```diff
@@ -1,25 +1,28 @@
 const express = require('express'),
       request = require('request'),
+      pug = require('pug'),
       morgan = require('morgan');
 
 var app = express();
 
+app.set('view engine', 'pug');
```
Comment (izelnakri) [inline index.js:8]: What does this do?

File: index.js (hunk 1 above)
Comment (Saidac) [reply index.js:25]: json file as string to be rendered as html?

File: index.js (hunk 2 above)
Comment (izelnakri) [reply index.js:19]: @Zahra120 Correct!

File: index.js (hunk 1 above)
Comment (izelnakri) [reply index.js:25]: @Saidac Correct! It is a json string

File: index.js (hunk 2 above)
Comment (kfungwm) [reply index.js:19]: dynamic for the users of github

File: index.js (hunk 3 above)
Comment (Saidac) [reply index.js:8]: we set the template engine to pug so we dont have to write it out every time as a file extension when we write out pathways to views

File: index.js (hunk 1 above)
Comment (kfungwm) [reply index.js:25]: show the content of json to html

File: index.js (hunk 3 above)
Comment (Zahra120) [reply index.js:8]: it enables us to use static template files in the application

Comment (izelnakri) [reply index.js:8]: answer: It allows us to use response.render()! When we set a view engine like this to an app then we dont need to put the extension in the path. Also, res.render() will lookup templates under /views directory by default.

**Pull request merged**